client/components/user/Settings/UserGeneral.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <v-form @submit.prevent="updateUser">
+    <user-avatar ref="userAvatar" :isEditing="isEditing" @editing="setEditing"/>
+    <v-layout class="general-info-container">
+      <v-flex class="fields-box">
+        <v-text-field
+          v-validate="{ required: true, email: true }"
+          v-model="context.email"
+          :error-messages="vErrors.collect('email')"
+          name="email"
+          label="Email"/>
+        <v-text-field
+          v-validate="{ max: 20 }"
+          v-model="context.firstName"
+          :error-messages="vErrors.collect('firstName')"
+          data-vv-as="first name"
+          name="firstName"
+          label="First name"/>
+        <v-text-field
+          v-validate="{ max: 20 }"
+          v-model="context.lastName"
+          :error-messages="vErrors.collect('lastName')"
+          data-vv-as="last name"
+          name="lastName"
+          label="Last name"/>
+      </v-flex>
+      <v-flex class="fields-box">
+        <v-layout class="role">
+          <v-icon color="primary">mdi-account-star</v-icon>
+          <h4 class="title">{{ user.role }}</h4>
+        </v-layout>
+        <v-text-field
+          v-validate="{ max: 50 }"
+          v-model="context.location"
+          :error-messages="vErrors.collect('location')"
+          name="location"
+          label="Location"
+          append-icon="mdi-map-marker"
+          outline/>
+      </v-flex>
+    </v-layout>
+    <v-card-actions>
+      <v-layout class="btn-actions">
+        <v-btn v-if="isEditing" @click="setEditing(false)" color="pink" flat>
```

Comment:
I suppose that it `isEditing` should be replaced with `hasChanges` since this way you display `Cancel` button only for image changes.